File: server/apps/departments/application/dto/department_dto.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class DepartmentDTO:

    department_id: str
    department_number: str
    name: str
    university_id: str
    university_name: str
    head_id: Optional[str]
    head_name: Optional[str]
    is_active: bool

File: client/src/application/departments/dto/department.dto.js
class DepartmentDTO {
  constructor({
    department_id,
    department_number,
    name,
    university_id,
    university_name,
    head_id,
    head_name,
    is_active,
  }) {
    this.departmentId = department_id;

    this.departmentNumber = department_number;

    this.name = name;

    this.universityId = university_id;

    this.universityName = university_name;

    this.headId = head_id ?? null;

    this.headName = head_name ?? null;

    this.isActive = is_active;
  }
}

module.exports = DepartmentDTO;
